src/application/boltz/claim.ts

```diff
@@ -0,0 +1,151 @@
+import type { OwnedInput } from 'liquidjs-lib';
+import {
+  Creator,
+  Updater,
+  Extractor,
+  Finalizer,
+  CreatorInput,
+  Transaction,
+  witnessStackToScriptWitness,
+  Pset,
+  ZKPValidator,
+  ZKPGenerator,
+  Blinder,
+  AssetHash,
+} from 'liquidjs-lib';
+import type { UnblindedOutput } from 'marina-provider';
+import type { Output } from 'liquidjs-lib/src/transaction';
+import { SignerService } from '../signer';
+import type { AppRepository, WalletRepository } from '../../domain/repository';
+import zkp from '@vulpemventures/secp256k1-zkp';
+
+/**
+ * Claim swaps
+ *
+ * @param utxos UTXOs that should be claimed or refunded
+ * @param witnessUtxo
+ * @param preimage
+ * @param redeemScript
+ * @param destinationScript the output script to which the funds should be sent
+ * @param fee how many satoshis should be paid as fee
+ * @param isRbf whether the transaction should signal full Replace-by-Fee
+ * @param assetHash asset hash of Liquid asset
+ * @param walletRepository
+ * @param appRepository
+ * @param password
+ * @param blindingKey blinding public key for the output; undefined if the output should not be blinded
+ * @param timeoutBlockHeight locktime of the transaction; only needed if the transaction is a refund
+ */
+export async function constructClaimTransaction(
+  utxos: UnblindedOutput[],
+  witnessUtxo: Output,
+  preimage: Buffer,
+  redeemScript: Buffer,
+  destinationScript: Buffer,
+  fee: number,
+  assetHash: string,
+  walletRepository: WalletRepository,
+  appRepository: AppRepository,
+  password: string,
+  isRbf = true,
+  blindingKey?: Buffer,
+  timeoutBlockHeight?: number
+): Promise<Transaction> {
+  const pset = Creator.newPset();
+  const updater = new Updater(pset);
+
+  let utxoValueSum = 0;
+
+  for (const [i, utxo] of utxos.entries()) {
+    utxoValueSum += utxo.blindingData?.value ?? 0;
+    pset.addInput(
+      new CreatorInput(
+        utxo.txid,
+        utxo.vout,
+        isRbf ? 0xfffffffd : 0xffffffff,
+        i === 0 ? timeoutBlockHeight : undefined
+      ).toPartialInput()
+    );
+    updater.addInSighashType(i, Transaction.SIGHASH_ALL);
+    updater.addInWitnessUtxo(i, witnessUtxo);
+    updater.addInWitnessScript(i, witnessUtxo.script);
+  }
+
+  /*
+  // Convert private blinding key to public blinding key
+  const masterBlindingKey = await walletRepository.getMasterBlindingKey();
+  if (!masterBlindingKey) throw new Error('Master blinding key not found');
+  const slip77 = SLIP77Factory(ecc);
+  const blindingKeyNode = slip77.fromMasterBlindingKey(blindingKey!);
+  const blindingPublicKey = blindingKeyNode.publicKey;
+  */
+
+  updater.addOutputs([
+    {
+      script: destinationScript,
+      blindingPublicKey: blindingKey,
+      // blindingPublicKey: blindingPublicKey,
```

Comment:
blinder needs at least one confidential output. I suggest you to pass confidential address as argument to the function. Then use the public key of that address here. Probably explain why blinding is failing